microindex/finder.go

```diff
@@ -166,34 +166,31 @@ func (f *Finder) Lookup(keys *zng.Record) (*zng.Record, error) {
 	return lookup(reader, compare, f.trailer.Order, eql)
 }
 
-func (f *Finder) LookupAll(ctx context.Context, hits chan<- *zng.Record, keys *zng.Record) error {
```

Comment:
@alfred-landrum after looking at the code for Finder I realized that it didn't make sense to have this return a zbuf.Reader since it would be confusing that the parent Finder would need to be closed once the Reader has been read through. As such I decided to keep the signature the same for Finder.LookupAll and added a separate FinderReader struct (implementing zbuf.ReadCloser) that can be used if one wants to stream results from a microindex search. @nwt had the idea that `LookupAll` could accept a zbuf.Writer, perhaps we'll want to change it to do this at some point.